### DISCUSSION TITLE
DS1302 not running during ESP8266 deep sleep

### DISCUSSION THREAD

**Discussion by Tobias4711** [Makuna/Rtc#195] [Q&A] [answered]
Dear members of the forum,

Thank you, Makuna, for creating the Rtc Library!

I am reaching out to you as I have encountered an issue and I am in need of your help and expertise. I am using Makuna's Rtc Library for my project. The problem is that my DS1302 real-time clock module is not functioning as expected when the ESP8266 enters deep sleep mode. The clock time stored on an SD card keeps resetting to the same value (time of compilation).

I have already investigated several possible causes and conducted relevant tests:

1. Faulty circuitry: I have built the circuit step by step and tested it after each stage. Everything works as intended without deep sleep, leading me to believe there is no issue with the circuit. The deep sleep functionality is only enabled through firmware.

2. Weak battery: I have tried three different battery manufacturers, using both new and non-expired batteries. The open-circuit voltage for all the batteries was 3.2V or higher, indicating sufficient voltage.

3. Setting the "clock halt" flag: My main concern is whether I understood correctly how to set the "clock halt" flag. In the program code, I used the following line:

rtc.SetIsRunning(true);

It is my understanding, that this instruction should correctly set the "clock halt" flag, allowing the clock to continue running during deep sleep. If I have made any mistakes here or if there are other potential reasons for the issue, I would greatly appreciate your input.

I would be very grateful for any brief feedback or assistance you can provide. Thank you in advance for your support!

Best regards,

Tobias

**Comment by Makuna** [comment, marked as answer]
You are correct, calling `rtc.SetIsRunning(true)` will turn ON the oscillator, and thus the RTC will run even when not powered by VCC, but must have a valid battery.  It sounds like you checked the battery already.   

Do you know if the battery you are using is a rechargeable?  LIR vs CR?  If it's a rechargeable (often used with the DS1302) then make sure the correct charging setup is used.  
You can check by calling `rtc.GetTrickleChargeSettings()`.  
If it's not a rechargeable, call `rtc.SetTrickleChargeSettings(DS1302Tcr_Disabled)`.  

Make a simple sketch (use an example) to set the clock within your circuit to the compile time.  Make another sketch that ONLY reads the time the RTC and upload it.   Disconnect from power for some time (at least a few minutes).  Then connect back to power and then serial output the date time and that should tell you if the RTC is keeping time when unpowered.

**Reply by Tobias4711** [reply]
Dear Makuna,

Thank you very much for your guidance and support. I wanted to share an update on my progress. Following your suggestions, I implemented the two small programs. Surprisingly, the DS1302 continues to count time even without external power. Encouraged by these results, I revisited the deep sleep issue. In the original program, the problem persisted. However, thanks to your recommendation, I took the simple program and expanded upon it. As a result, the clock now continues to run during deep sleep. I will now proceed to gradually expand the program (writing to SD card already works as an extension). My current suspicion is that I might have introduced something in the program code in an order that doesn't work when using deep sleep.

Once again, thank you for your invaluable assistance. Your insights have been instrumental in resolving this issue.

Best regards,
Tobias
